src/main/java/com/redhat/pantheon/servlet/AsciidocRenderingServlet.java

```diff
@@ -110,8 +115,14 @@ private Content generateHtml(SlingHttpServletRequest request, Resource resource)
         RubyInstanceConfig config = new RubyInstanceConfig();
         config.setLoader(Thread.currentThread().getContextClassLoader());
 
-        Asciidoctor instance = Asciidoctor.Factory.create(
-                singletonList("uri:classloader:/gems/asciidoctor-1.5.8/lib"));
+        Enumeration<URL> gems = PantheonBundleActivator.getContext().getBundle().findEntries("gems", "*", false);
```

Comment:
Why did we have to add these lines? I mean, this is about finding the ruby gems for asciidoctorj to work, which we weren't having a problem with before.

Reply:
It's because the templating functionality required the presence of additional gems. In particular, we needed tilt, haml, and thread_safe, which luckily were all already included in the jar (but we hadn't needed to point at them before, so they were being ignored).

So rather than hardcoding a bunch of directories which are likely going to change if we move to a newer version of asciidoctorj, I tried to make it somewhat dynamic. (There are actually 10 different gems directories in the JAR! Now we include them all.)

But there's still the assumption that the directories will take the form of /gems/xxxxx/lib. Hopefully that's safe but I'm open to better ideas.

Reply:
Got it. 
We just need to take it out of this class. Sounds like a one time setup thing for asciidoctorj.